modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingFlowComposable.kt

```diff
@@ -26,192 +26,219 @@ fun OnboardingFlowComposable(
     completeOnboardingToDiscover: () -> Unit,
     signInState: SignInState
 ) {
-    AppThemeWithBackground(theme) {
-        val navController = rememberNavController()
-
-        val startDestination = when (flow) {
-            OnboardingFlow.LoggedOut,
-            is OnboardingFlow.PlusAccountUpgradeNeedsLogin,
-            OnboardingFlow.InitialOnboarding -> OnboardingNavRoute.logInOrSignUp
-
-            // Cannot use OnboardingNavRoute.PlusUpgrade.routeWithSource here, it is set as a defaultValue in the PlusUpgrade composable,
-            // see https://stackoverflow.com/a/70410872/1910286
-            is OnboardingFlow.PlusAccountUpgrade, // FIXME this should just open the purchase modal
-            is OnboardingFlow.PlusFlow -> OnboardingNavRoute.PlusUpgrade.route
+    if (flow is OnboardingFlow.PlusAccountUpgrade) {
+        Content(
+            theme = theme,
+            flow = flow,
+            exitOnboarding = exitOnboarding,
+            completeOnboardingToDiscover = completeOnboardingToDiscover,
+            signInState = signInState
+        )
+    } else {
+        AppThemeWithBackground(theme) {
+            Content(
+                theme = theme,
+                flow = flow,
+                exitOnboarding = exitOnboarding,
+                completeOnboardingToDiscover = completeOnboardingToDiscover,
+                signInState = signInState
+            )
         }
+    }
+}
 
-        val onAccountCreated = {
-            navController.navigate(OnboardingRecommendationsFlow.route) {
-                // clear backstack after account is created
-                popUpTo(OnboardingNavRoute.logInOrSignUp) {
-                    inclusive = true
-                }
+@Composable
+private fun Content(
+    theme: Theme.ThemeType,
+    flow: OnboardingFlow,
+    exitOnboarding: () -> Unit,
+    completeOnboardingToDiscover: () -> Unit,
+    signInState: SignInState
+) {
+    val navController = rememberNavController()
+
+    val startDestination = when (flow) {
+        OnboardingFlow.LoggedOut,
+        is OnboardingFlow.PlusAccountUpgradeNeedsLogin,
+        OnboardingFlow.InitialOnboarding -> OnboardingNavRoute.logInOrSignUp
+
+        // Cannot use OnboardingNavRoute.PlusUpgrade.routeWithSource here, it is set as a defaultValue in the PlusUpgrade composable,
+        // see https://stackoverflow.com/a/70410872/1910286
+        is OnboardingFlow.PlusAccountUpgrade, // FIXME this should just open the purchase modal
```

Comment:
In order to enable just the pricing modal, I excluded the `OnboardingUpgradeFeaturesPage` content in the `PlusAccountUpgrade` flow and made the onboarding activity transparent. This method successfully worked even on an API-level 24 emulator with hardcoded feature cards and upgrade buttons. But please let me know if you encounter any issues or if it interferes with any other flow.